lib/document.js

```diff
@@ -454,6 +455,22 @@ function $__applyDefaults(doc, fields, skipId, exclude, hasIncludedChildren, isB
     }
   }
 }
+/**
+ * Checks which paths have defaults
+ * to then delete them from the doc
+ *
+ * @param {Object} doc
+ */
+function $__unapplyDefaults(doc) {
+  let values = Object.values(doc.$__schema.paths);
+  for (let i = 0; i < values.length; i++) {
+    if (values[i].defaultValue && values[i].instance != 'ObjectID') {
```

Comment:
You should use`$isDefault()` to check if a value is set to its default value.

Reply:
https://mongoosejs.com/docs/api/document.html#document_Document-$isDefault

Reply:
Object.values is unfortunately incompatible with older versions of node so I have updated the current method.

Reply:
will do

Reply:
I'm not exactly sure how to use it. I do `doc.$isDefault('name')` which returns false even though it should return true.